### PR TITLE
Feat spectrogram factory inputs

### LINKED ISSUE #52: Scope what inputs Spectrogram factory is expected to accept?
Ideally
 * files - str, path, glob - fits, cdf, dat, custom
 * urls
 * data, header pairs?
 * ???

So really looks like need something almost identical to what's in `MapFactory` but  seems like would be effectively be copy pasting a lot which is bad .... ?

### PR DESCRIPTION
**Pull request by samaloney**
### Description

This PR makes the spectrogram factory much more like its sunpy counterparts in map and timeseries.

* Now handles many different inputs, path, strings, data-header pairs, urls
* Also added a new client and spectrogram class for WIND/WAVES 

Fixes #52

### CODE REVIEW

File: radiospectra/net/sources/wind.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            # min and max of combined lft and hfr contained in give wavelength range
+            if a.Wavelength(RECEIVER_FREQUENCIES['rad1'].min,
+                            RECEIVER_FREQUENCIES['rad2'].max) in wavelength:
+                receivers = ['rad1', 'rad2']
+            # If we get here the is no overlap so set to empty list
+        return receivers
+
+    def search(self, *args, **kwargs):
+        """
+         Query this client for a list of results.
```
Comment (nabobalis) [inline radiospectra/net/sources/wind.py:90]: This all seems to have one extra space for its indentation.

File: radiospectra/spectrogram2/sources.py [hunk 2]
```diff
@@ -93,31 +145,58 @@ def observatory_location(self):
...
+    --------
+    >>> import radiospectra.net
+    >>> from sunpy.net import Fido, attrs as a
+    >>> from radiospectra.spectrogram2 import Spectrogram
+    >>> query = Fido.search(a.Time('2021/05/07 00:00', '2021/05/07 23:00'), a.Instrument.eovsa)  #doctest: +REMOTE_DATA
+    >>> downloaded = Fido.fetch(query[0][0])  #doctest: +REMOTE_DATA
+    >>> spec = Spectrogram(downloaded[0])  #doctest: +REMOTE_DATA
+    >>> spec  #doctest: +REMOTE_DATA
+    <EOVSASpectrogram OWENS VALLEY, EOVSA, EOVSA 1105371.117591858 kHz - 17979686.737060547 kHz, 2021-05-07T13:48:20.999 to 2021-05-08T01:50:59.999>
+    >>> spec.plot()  #doctest: +REMOTE_DATA
```
Comment (nabobalis) [inline radiospectra/spectrogram2/sources.py:166]: Do these show up in the final doc build?

File: radiospectra/spectrogram2/spectrogram.py [hunk 3]
```diff
@@ -1,26 +1,294 @@
...
+        """
+        Plot the spectrogram.
+
+        Parameters
+        ----------
+        axes : `matplotlib.axis.Axes`, optional
+            The axes where the plot will be added.
+        kwargs :
+            Arguments pass to the plot call `pcolormesh`.
+        Returns
```
Comment (nabobalis) [inline radiospectra/spectrogram2/spectrogram.py:147]:
```suggestion

        Returns
```

File: radiospectra/spectrogram2/spectrogram.py [hunk 4]
```diff
@@ -37,31 +305,210 @@ class SpectrogramFactory(BasicRegistrationFactory):
...
+        * tuples of data,header
+        * data, header not in a tuple
+        * data, wcs object in a tuple
+        * data, wcs object not in a tuple
+        * filename, as a str or pathlib.Path, which will be read
+        * directory, as a str or pathlib.Path, from which all files will be read
+        * glob, from which all files will be read
+        * url, which will be downloaded and read
+        * lists containing any of the above.
+        Example
```
Comment (nabobalis) [inline radiospectra/spectrogram2/spectrogram.py:332]:
```suggestion

        Example
```

File: radiospectra/spectrogram2/spectrogram.py [hunk 5]
```diff
@@ -37,31 +305,210 @@ class SpectrogramFactory(BasicRegistrationFactory):
...
+        function and to the constructor for the final type. For spectrogram types,
+        validation function must take a data-header pair as an argument.
 
-    def _read_file(self, file):
+        Parameters
+        ----------
+        silence_errors : `bool`, optional
+            If set, ignore data-header pairs which cause an exception.
+            Default is ``False``.
+        Notes
```
Comment (nabobalis) [inline radiospectra/spectrogram2/spectrogram.py:426]:
```suggestion

        Notes
```

File: radiospectra/spectrogram2/spectrogram.py [hunk 6]
```diff
@@ -37,31 +305,210 @@ class SpectrogramFactory(BasicRegistrationFactory):
...
+        return new_maps
+
+    # def __call__(self, *args, **kwargs):
+    #     if len(args) == 1 and not kwargs:
+    #         arg = args[0]
+    #         if isinstance(arg, (str, Path)):
+    #             file = Path(arg)
+    #             meta, data = self._read_file(file)
+    #
+    #     return self._check_registered_widget(meta=meta, data=data, **kwargs)
```
Comment (nabobalis) [inline radiospectra/spectrogram2/spectrogram.py:468]:
```suggestion
```

**Review by nabobalis — APPROVED** [review]
A few minor comments, but nothing I want to block the PR on.

File: radiospectra/spectrogram2/sources.py (hunk 2 above)
Comment (samaloney) [reply radiospectra/spectrogram2/sources.py:166]: The plot does not but the rest does.

Comment (samaloney) [reply radiospectra/spectrogram2/sources.py:166]: Fixed

Comment (nabobalis) [reply radiospectra/spectrogram2/sources.py:166]: In sunpy we skip them, I am unsure why.

**Pull request merged**